Tests/js/models/assets/ez-locationmodel-tests.js

```diff
@@ -56,7 +57,99 @@ YUI.add('ez-locationmodel-tests', function (Y) {
 
     }));
 
+    trashTest = new Y.Test.Case({
+        name: "eZ Location Model trash tests",
+
+        setUp: function () {
+            this.model = new Y.eZ.Location();
+        },
+
+        tearDown: function () {
+            this.model.destroy();
+            delete this.model;
+        },
+
+        "Should move location to trash": function () {
+            var capiMock = new Mock(),
+                contentServiceMock = new Mock(),
+                locationId = 'ronaldo-luis-nazario-de-lima',
+                loadRootResponse = {
+                    "document": {
+                        "Root": {
+                            "trash": {
+                                "_href": 'gianfranco-zola'
+                            }
+                        }
+                    }
+                };
+
+            Mock.expect(capiMock, {
+                method: 'getContentService',
+                returns: contentServiceMock,
+            });
+
+            Mock.expect(contentServiceMock, {
+                method: 'loadRoot',
+                args: [Mock.Value.Function],
+                run: function (cb) {
+                    cb(false, loadRootResponse);
+                }
+            });
+
+            Mock.expect(this.model, {
+                method: 'get',
+                args: [Mock.Value.String],
+                returns: locationId
+            });
+
+            Mock.expect(contentServiceMock, {
+                method: 'moveSubtree',
+                args: [Mock.Value.String, Mock.Value.String, Mock.Value.Function],
```

Comment:
this is a case where the arguments are important, so here instead of `Mock.Value.String` you should put the actual expected values (`locationId`  and `loadRootResponse.document.Root.trash._href`).